File: go/1657.go
package main

import "sort"

func closeStrings(word1 string, word2 string) bool {
	countA := make([]int, 26)
	countB := make([]int, 26)
	for _, word := range word1 {
		countA[word-'a']++
	}
	for _, word := range word2 {
		countB[word-'a']++
	}
	for i := range countA {
		if countA[i] > 0 && countB[i] == 0 {
			return false
		}
		if countB[i] > 0 && countA[i] == 0 {
			return false
		}
	}
	sort.Ints(countA)
	sort.Ints(countB)
	for i := range countA {
		if countA[i] != countB[i] {
			return false
		}
	}
	return true
}

func main() {

}

File: python/1657.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) != len(word2):
            return False
        setA = set()
        setB = set()
        countA = [0] * 26
        countB = [0] * 26
        for word in word1:
            setA.add(word)
            countA[ord(word) - ord('a')] += 1
        for word in word2:
            setB.add(word)
            countB[ord(word) - ord('a')] += 1
        if setA != setB:
            return False
        countA.sort()
        countB.sort()
        for i in range(len(countB)):
            if countB[i] != countA[i]:
                return False
        return True